test/FunctionalTests/Microsoft.OData.Core.Tests/JsonLight/ODataJsonLightPropertySerializerTests.cs

```diff
@@ -577,15 +579,55 @@ public void WritingPropertyShouldWriteCollectionResourceInstanceAnnotation()
                 result);
         }
 
+#if NETCOREAPP3_1_OR_GREATER
+        [Fact]
+        public void WritingJsonElementProperties_ShouldSerializeJsonInput()
+        {
+            string jsonInputString = "{\"foo\":\"bar\"}";
+            var jsonDocument = System.Text.Json.JsonDocument.Parse(jsonInputString);
+
+            var property = new ODataProperty()
+            {
+                Name = "JsonProp",
+                Value = new ODataJsonElementValue(jsonDocument.RootElement)
+            };
+
+            var result = SerializeProperty(null, property);
+            Assert.Equal("{\"JsonProp\":{\"foo\":\"bar\"}}", result);
+        }
+
+        [Fact]
+        public void WritingJsonElementProperties_ShouldSerializeJsonInput_WithODataUtf8JsonWriter()
+        {
+            string jsonInputString = "{\"foo\":\"bar\"}";
+            var jsonDocument = System.Text.Json.JsonDocument.Parse(jsonInputString);
+
+            var property = new ODataProperty()
+            {
+                Name = "JsonProp",
+                Value = new ODataJsonElementValue(jsonDocument.RootElement)
+            };
+
+            Action<IContainerBuilder> configureServices = (IContainerBuilder builder) =>
+            {
+                builder.AddService<IStreamBasedJsonWriterFactory>(ServiceLifetime.Singleton, _ => DefaultStreamBasedJsonWriterFactory.Default);
+            };
+
+            var result = SerializeProperty(null, property, configureServices);
+            Assert.Equal("{\"JsonProp\":{\"foo\":\"bar\"}}", result);
+        }
+#endif

```

Comment:
In the `ODataValueUtils`, there's this block:
```csharp
if (objectToConvert is JsonElement jsonElement)
{
    return new ODataJsonElementValue(jsonElement);
}
```
Do you have a test/tests for that scenario?

Reply:
I had not added tests for this, thanks for the catch. Let me add them.

I've just looked at the description of the `ODataValueUtiles.FromODataValue()` method and have decided to remove the special handling for `JsonElement` from that method. It seems like that method was only intended to unwrap primitive and null values. However, I'll keep the `JsonElement` handling in `ToODataValue()` and add relevant tests.